arbnode/batch_poster.go

```diff
@@ -96,28 +103,38 @@ var TestBatchPosterConfig = BatchPosterConfig{
 	MaxBatchPostInterval: 0,
 	CompressionLevel:     2,
 	DASRetentionPeriod:   time.Hour * 24 * 15,
-	HighGasThreshold:     0.,
 	HighGasDelay:         0,
 	GasRefunderAddress:   "",
+	DataPoster:           dataposter.TestDataPosterConfig,
 }
 
 func NewBatchPoster(l1Reader *headerreader.HeaderReader, inbox *InboxTracker, streamer *TransactionStreamer, config *BatchPosterConfig, contractAddress common.Address, transactOpts *bind.TransactOpts, das das.DataAvailabilityService) (*BatchPoster, error) {
-	inboxContract, err := bridgegen.NewSequencerInbox(contractAddress, l1Reader.Client())
+	seqInbox, err := bridgegen.NewSequencerInbox(contractAddress, l1Reader.Client())
 	if err != nil {
 		return nil, err
 	}
 	if len(config.GasRefunderAddress) > 0 && !common.IsHexAddress(config.GasRefunderAddress) {
 		return nil, fmt.Errorf("invalid gas refunder address \"%v\"", config.GasRefunderAddress)
 	}
+	seqInboxABI, err := bridgegen.SequencerInboxMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	dataPoster, err := dataposter.NewDataPoster[batchPosterPosition](l1Reader, transactOpts, &config.DataPoster)
```

Comment:
better if the dataposter is held/created directly in Node and newbatchposter (builder?) gets a pointer.

Reply:
if possible